js/components/main.js

```diff
@@ -715,6 +715,10 @@ class Main extends ImmutableComponent {
   }
 
   get allSiteSettings () {
+    const activeFrame = FrameStateUtil.getActiveFrame(this.props.windowState)
+    if (activeFrame && activeFrame.get('isPrivate')) {
+      return this.props.appState.get('temporarySiteSettings')
+    }
```

Comment:
need to be careful here that the `allSiteSettings` frame property is consistent with what is being sent in `js/state/contentSettings.js` and also `registerPermissionHandler` in `app/filtering.js`. the expected behavior is that private context inheirit settings from regular contexts but not vice-versa, so I think what you want here is a merge of `temporarySiteSettings` and `siteSettings` for private frames.